Add new database source loader with commit after file promoted

MikeHeyes opened the pull request:
Add new loader duplicating DatabaseSource but with a commit after all sql statements in the file have been executed

MikeHeyes committed:
Add new loader duplicating DatabaseSource but with a commit after every sql statement in the file has been executed

benbasson commented:
It looks to me like `DatabaseSource `and `DatabaseSourceWithCommitLoader` are identical apart from the commit. I would suggest that it would be cleaner to abstract away the functionality in `doPromote` and then have `DatabaseSource` extend `DatabaseSourceWithCommitLoader` and override `doPromote` to do the commit.

AledLewis commented:
@benbasson had an offline chat.
You're right, it's essentially a one-line change. I've asked Mike to:
1. Rename unsafe commit to implicit commit.
2. Add a boolean constructor to `DatabaseSource` switching on implicit commits or not (don't think the time's now to start to add strategies etc.)

MikeHeyes committed:
Replace DatabaseSourceLoaderWithCommit with a simple constructor switch
on the original DatabaseSourceLoader class. Rename loader tag for
builder to DatabaseSourceImplicitCommit

AledLewis commented:
Thanks Mike!
